python/ray/data/impl/stats.py

```diff
@@ -0,0 +1,296 @@
+from contextlib import contextmanager
+from typing import List, Optional, Dict, Set, Tuple
+import time
+import collections
+import numpy as np
+
+import ray
+from ray.data.block import BlockMetadata
+from ray.data.impl.block_list import BlockList
+
+
+def fmt(seconds: float) -> str:
+    if seconds > 1:
+        return str(round(seconds, 2)) + "s"
+    elif seconds > 0.001:
+        return str(round(seconds * 1000, 2)) + "ms"
+    else:
+        return str(round(seconds * 1000 * 1000, 2)) + "us"
+
+
+class Timer:
```

Comment:
there seems to be something equivalent in https://sourcegraph.com/github.com/ray-project/ray/-/blob/python/ray/util/timer.py?L5:7

Reply:
That one seems a bit different (more like a window gauge).